### ISSUE TITLE
EDM editor: components looks bad

### ISSUE THREAD

**Issue by Uafinder** [Kademi/kademi-dev#5012]
As discussed with Brad should be improved the initial view for the component (when it's not configured)
![image](https://user-images.githubusercontent.com/1945230/36424049-be756be8-164a-11e8-85d7-d474420a4ea3.png)
http://vladtest35a.admin.kademi-ci.co/groupEmails/6367639/#sendInfo-tab

brad 10:34 AM
Also, the blog list component looks HORRIBLE when its not configured. can you also log an issue for that?

same thing with Points Expiring Component
![image](https://user-images.githubusercontent.com/1945230/36430626-4d30821a-165e-11e8-9fdf-1a335f4724b9.png)
initial view for the component are poor

same for Points Leaderboard component
![image](https://user-images.githubusercontent.com/1945230/36431057-4afd051c-165f-11e8-8fab-2e9e226e8f5f.png)

**Comment by bradmac** [comment]
Yep, please make sure components have the same structure (ie the height, surrounding div layout) as they will have when correctly configured.

Otherwise users cant actually do visual editing

**Comment by bradmac** [comment]
Done

**Comment by Uafinder** [comment]
Also for Single product component
![image](https://user-images.githubusercontent.com/1945230/36792228-a6e1ee44-1ca2-11e8-8687-6671ed82cc34.png)
Leaderboard component as well
![image](https://user-images.githubusercontent.com/1945230/36792297-e5036a0e-1ca2-11e8-898d-e5b7700d702e.png)
http://vladtest38.admin.kademi-ci.co/

**Comment by bradmac** [comment]
hey @ducdhm can you please follow a similar approach to what i did with blogArticleList component?

- layout should be coded once
- check for correct configuration in the component template where its being used. If no config or its invalid generate alternative content

**Comment by ducdhm** [comment]
Fixed!

**Comment by Uafinder** [comment]
ok, those ones which I provide earlier is ok
three more EDM components with the same problem
![image](https://user-images.githubusercontent.com/1945230/36892222-bbbd23c2-1e0c-11e8-8486-71c0f0e789d0.png)

1. KPI leaderboard
2. product list component
3. reward store categories list

Also please note that
KPI leaderboard component and Reward store category component settings has incorrect names
![image](https://user-images.githubusercontent.com/1945230/36892365-2d7a3658-1e0d-11e8-9399-01360978c778.png)
![image](https://user-images.githubusercontent.com/1945230/36892390-42574fe8-1e0d-11e8-9c72-4600927575ff.png)

**Comment by ducdhm** [comment]
Done!

**Comment by Uafinder** [comment]
Yeah, looks fine at the moment

**Comment by Uafinder** [comment]
Some more component:
1. Survey component
![image](https://user-images.githubusercontent.com/1945230/37333090-19f263ce-26b1-11e8-8cd7-d5499aab5073.png)
http://vladtest40e.admin.kademi-ci.co/groupEmails/9703122/#messages-tab

**Comment by anhnt** [comment]
Updated survey email

**Comment by Uafinder** [comment]
Nope, still active
![image](https://user-images.githubusercontent.com/1945230/37523070-8682fe66-292e-11e8-877b-87eb41dce697.png)
Fresh account
http://vladtest42a.admin.kademi-ci.co/groupEmails/10428722/#sendInfo-tab

**Comment by anhnt** [comment]
Please update ksurvey app to version 1.3.2

**Comment by bradmac** [comment]
Hey guys, we should never rely on manually changing versions.

@Uafinder when you re-test you need to do it with a fresh account, otherwise you will always be testing with the old version of the libs which will still have the problem.

@anhnt as the developer its your responsibility to ensure your change is published, so it will get used in a fresh account

**Comment by Uafinder** [comment]
Understood
Looks better

**Comment by Uafinder** [comment]
Some more components
![image](https://user-images.githubusercontent.com/1945230/37964085-f1632a92-31c8-11e8-95be-b79991b6dfed.png)
- calendar event component
- promotions component
- leaderboard KPI component
http://vladtest46b.admin.kademi-ci.co/groupEmails/13781319/#messages-tab
